modin/pandas/general.py

```diff
@@ -454,6 +462,18 @@ def concat(
         ).index
     else:
         new_idx = None
+
+    if len(list_of_objs) == 0:
+        idx = input_list_of_objs[0].index.append(
+            [f.index for f in input_list_of_objs[1:]]
+        )
+        if axis == 1 or any(
+            isinstance(obj, (DataFrame, pandas.DataFrame)) for obj in input_list_of_objs
+        ):
+            return DataFrame(index=idx)
+        else:
+            return Series(index=idx)
```

Comment:
It looks like a correction of the consequences of the fact that those objects were filtered out, information about which we need in the future. Let's try to avoid it.

Reply:
If all objects are filtered out, we need concatenate indexes and return either an empty DataFrame or Series. According to the documentation:


When concatenating all Series along the index (axis=0), a Series is returned. When objs contains at least one DataFrame, a DataFrame is returned. When concatenating along the columns (axis=1), a DataFrame is returned.